test/test_torch.py

```diff
@@ -2133,22 +2134,26 @@ def test_elastic_state(self):
     def test_timeline_api(self):
         hvd.init()
 
-        def check_file(fname):
+        def check_file(fname, check_cycle=True):
             if hvd.rank() == 0:
                 with open(fname, 'r') as timeline_file:
                     timeline_text = timeline_file.read()
+                    print("Timeline text is : ")
+                    print(timeline_text + "\n")
                     assert 'allreduce.test_allreduce' in timeline_text, timeline_text
                     assert 'NEGOTIATE_ALLREDUCE' in timeline_text, timeline_text
                     assert 'ALLREDUCE' in timeline_text, timeline_text
-                    assert 'CYCLE_START' in timeline_text, timeline_text
+                    if check_cycle is True:
+                        assert 'CYCLE_START' in timeline_text, timeline_text
 
         with temppath() as fname1:
             hvd.start_timeline(fname1, mark_cycles=True)
-
-            # Perform a simple allreduce operation
+            print("Calling all reduce\n")
             hvd.allreduce(torch.tensor([1, 2, 3], dtype=torch.float32), name='test_allreduce')
-
-            # Wait for timeline to drain before reading it
+            print("Calling stop timeline\n")
+            # stop timeline will immediately stop events to be registered in timeline. We are providing some time
+            # before calling stop so that events can be registered in timeline file.
+            time.sleep(0.2)
```

Comment:
When we call Start_timeline, timeline is not started immediately and it is started in next cycle of background thread loop(to avoid races and big locks etc.). I put sleep as a caution to make sure that tests remains deterministic.

Reply:
I think we should make these calls blocking.  Should be simple enough to do by capturing the change of the pending flag.  Does that sounds reasonable to you?

Reply:
Done! Made both start_timeline and stop_timeline calls blocking.